tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15176,6 +15176,83 @@ QualType Sema::getHLSLDefaultSpecialization(TemplateDecl *Decl) {
   return QualType();
 }
 
+static bool isRelatedDeclMarkedNointerpolation(Expr *E) {
+  if (!E)
+    return false;
+  E = E->IgnoreCasts();
+  if (auto *DRE = dyn_cast<DeclRefExpr>(E))
+    return DRE->getDecl()->hasAttr<HLSLNoInterpolationAttr>();
+
+  if (auto *ME = dyn_cast<MemberExpr>(E))
+    return ME->getMemberDecl()->hasAttr<HLSLNoInterpolationAttr>() ||
+           isRelatedDeclMarkedNointerpolation(ME->getBase());
+
+  if (auto *HVE = dyn_cast<HLSLVectorElementExpr>(E))
+    return isRelatedDeclMarkedNointerpolation(HVE->getBase());
+
+  if (auto *ASE = dyn_cast<ArraySubscriptExpr>(E))
+    return isRelatedDeclMarkedNointerpolation(ASE->getBase());
+
+  return false;
+}
+
+static bool CheckIntrinsicGetAttributeAtVertex(Sema *S, FunctionDecl *FDecl,
+                                               CallExpr *TheCall) {
+  assert(TheCall->getNumArgs() > 0);
+  auto argument = TheCall->getArg(0)->IgnoreCasts();
+
+  if (!isRelatedDeclMarkedNointerpolation(argument)) {
+    S->Diag(argument->getExprLoc(), diag::err_hlsl_parameter_requires_attribute)
+        << 0 << FDecl->getName() << "nointerpolation";
+    return true;
+  }
+
+  return false;
+}
+
+bool Sema::CheckHLSLIntrinsicCall(FunctionDecl *FDecl, CallExpr *TheCall) {
+  auto attr = FDecl->getAttr<HLSLIntrinsicAttr>();
+
+  switch (hlsl::IntrinsicOp(attr->getOpcode())) {
+  case hlsl::IntrinsicOp::IOP_GetAttributeAtVertex:
+    // See #hlsl-specs/issues/181. Feature is broken. For SPIR-V we want
+    // to limit the scope, and fail gracefully in some cases.
+    if (!getLangOpts().SPIRV)
+      return false;
+    if (FDecl->getName() != "GetAttributeAtVertex")
```

Comment:
Is this possible? I assume if the intrinsic op is set it should be this function.

Reply:
It just for a weird test in DXC, if an extension registers an instrinsic and there is a name conflict.

Reply:
I'm confused. Is there a test case where this fails? If we've assigned an intrinsic op then we're telling the compiler to treat it as that intrinsic regardless of the name of the function. I don't see anywhere else in DXC where we also separately need to verify the name of the function.

Reply:
IIRC the test was `ExtensionTest.EvalAttributeCollision`, which seems to register an intrinsic (collide_proc), with the same opcode as a registered intrinsic.
So here I hit the condition because the opcodes are the same, but the name mismatch.

Reply:
I see the test now. @adam-yang wrote it. I'm unsure what the context there is (although it terrifies me).

It shouldn't actually matter here though since that is a DXIL-only test and right above here you return if you're targeting DXIL, so this code shouldn't be possible to hit. We could make this an assert if you're concerned that it might come up for SPIR-V, but I think the inline SPIR-V proposal provides a more sane way to expose extensions than inserting a table of overridden intrinsics.

Reply:
Of course, forgot I gated this on `isSPIRV`. Cleaning that up now, adding an assert.